### PR TITLE
Ic sheet9 regularization

### PR DESCRIPTION
**Pull request by patriciomassaro**

### CODE REVIEW

**Comment by patriciomassaro** [comment]
[IC exercise sheet 9](https://trello.com/c/FTh8zreh/247-ic-exercise-sheet-9)

File: exercises/regularization/ex_rnw/ex_lasso_regularization.Rnw [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+		\lz
+		\lz
+	
+	
+	
+	\item 
+	
+		Note that the minimization problem on the right-hand side of (i) can be written as $\sum_{i=1}^p g_i(\theta_i),$ where 
+		$$ 
+			g_i(\theta) =  - \thetah_{i}\theta + \frac{\theta^2}{2} + \lambda |\theta|.
```
Comment (ludwigbothmann) [inline exercises/regularization/ex_rnw/ex_lasso_regularization.Rnw:43]: Perhaps g_i(\theta_i)? Is not wrong as-is, but perhaps clearer?

File: exercises/regularization/ex_rnw/sol_lasso_regularization.Rnw [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+                \thetabh &= \left(\underbrace{{\Xmat}^T \Xmat}_{\id}\right)^{-1} \Xmat^T\yv \\
+                \thetabh &= \Xmat^T \yv  
+            \end{aligned}
+        \end{equation}
+
+        We will now use the result of eq. \ref{eq:solve_b} to show that:
+        \begin{equation}
+            \begin{aligned}
+                \argmin_{\thetab}  \riskrt &= \argmin_{\thetab} \frac12 \sum_{i=1}^n \left( \yi - \thetab^\top \xi \right)^2 + \lambda \sum_{i=1}^p  |\theta_i| \\
+                &= \argmin_{\thetab}  \frac12  \| \yv - \xv \thetab  \|_2^2 + \lambda \|\thetab\|_1 \\
```
Comment (ludwigbothmann) [inline exercises/regularization/ex_rnw/sol_lasso_regularization.Rnw:18]: Here and later: Capital X, i.e., matrix

File: exercises/regularization/ex_rnw/sol_lasso_regularization.Rnw [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+                &= \argmin_{\thetab} \frac12 ( \yv - \xv \thetab)^T ( \yv - \xv \thetab) + \lambda \|\thetab\|_1 \\ 
+                &= \argmin_{\thetab}  \frac12 \left(\underbrace{\yv^T \yv}_{indep. \  of \  \thetab} - 2 \underbrace{\yv^T \xv}_{\thetab^T} \theta + \theta^T \underbrace{\xv^T \xv}_{\id} \thetab \right) + \lambda \|\thetab\|_1 \\
```
Comment (ludwigbothmann) [inline exercises/regularization/ex_rnw/sol_lasso_regularization.Rnw:20]: Some thetas are not bold right now.

File: exercises/regularization/ex_rnw/ex_lasso_regularization.Rnw [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+	\riskrt = \risket + \lambda \|\thetab\|_1 =  \frac12 \sum_{i=1}^n \left( \yi - \thetab^\top \xi \right)^2 + \lambda \sum_{i=1}^p  |\theta_i|.
+$$
+Assume that ${\Xmat}^T \Xmat = \id,$ which holds if $\Xmat$ has orthonormal columns. Show that the minimizer $\thetah_{\text{Lasso}} = (\thetah_{\text{Lasso},1},\ldots,\thetah_{\text{Lasso},p})^\top$ is given by
+
+$$	
+	\thetah_{\text{Lasso},i} = sgn(\thetah_{i}) \max\{|\thetah_{i}| - \lambda,0 \}, \quad i=1,\ldots,p,	
+$$
+
+where $\thetabh = (\thetah_{1},\ldots,\thetah_{p})^\top = ({\Xmat}^T \Xmat)^{-1} \Xmat^T\yv$ is the minimizer of the unregularized empirical risk (w.r.t.\ the L2 loss). For this purpose, use the following steps:
+\begin{itemize}
```
Comment (ludwigbothmann) [inline exercises/regularization/ex_rnw/ex_lasso_regularization.Rnw:19]: - Please use enumerate
- use the same symbols as you refer to them in later parts (there you write iii) and iv) - on the solution it is a, b,c, => should be consistent)

**Review by ludwigbothmann — CHANGES_REQUESTED** [review]

File: exercises/regularization/ex_rnw/ex_lasso_regularization.Rnw (hunk 1 above)
Comment (patriciomassaro) [reply exercises/regularization/ex_rnw/ex_lasso_regularization.Rnw]: Changed

**Review by ludwigbothmann — APPROVED** [review]

**Pull request merged**